pkg/app/piped/executor/analysis/analysis.go

```diff
@@ -146,7 +175,11 @@ func (e *Executor) Execute(sig executor.StopSignal) model.StageStatus {
 		return model.StageStatus_STAGE_FAILURE
 	}
 
-	status := executor.DetermineStageStatus(sig.Signal(), e.Stage.Status, model.StageStatus_STAGE_SUCCESS)
+	if status != model.StageStatus_STAGE_NOT_STARTED_YET {
+		return status
+	}
```

Comment:
why do we need this? if the condition is `==` I can guess it means skip determining stage status in case of not yet started stage, but if it is not, I guess this overlaps with the thing we do in the next line (L182) - determining the status of stage regarding the conditions. If this is necessary, add comment for this would be nice and easier to understand

Reply:
I got your points and addressed them by https://github.com/pipe-cd/pipecd/pull/3525/commits/ed3fc5577cabcc713aac3f95cf7855dacedb01c1